src/odemis/acq/stream/_sync.py

```diff
@@ -918,10 +934,10 @@ def _runAcquisitionEbeam(self, future):
             list_polarizations = [None]
 
             # check if polarization VA exists, overwrite list of polarization value
-            if model.hasVA(self.streams[-1], "polarization"):
+            if self._analyzer:
                 if self._polarization.value == POL_6POS:
-                    # hardcode list of polarization positions necessary for polarimetry analysis
-                    list_polarizations = ['horizontal', 'vertical', 'posdiag', 'negdiag', 'rhc', 'lhc']
+                    # use hardcode list of polarization positions necessary for polarimetry analysis
```

Comment:
Please,  copy (or move) this comment to the line where the `POL_POSITIONS` is defined.

BTW, list_polarizations doesn't need to be a _list_, a tuple would be fine. Maybe just rename the the variable and directly use the constant (without copying it into a list).